sql/core_clients_last_seen_raw_v1.sql

```diff
@@ -0,0 +1,63 @@
+  -- Equivalent to, but more efficient than, calling udf_bitmask_range(1, 28)
+CREATE TEMP FUNCTION
+  udf_bitmask_lowest_28() AS (0x0FFFFFFF);
+  -- Equivalent to, but more efficient than, calling udf_bitmask_range(1, 7)
+CREATE TEMP FUNCTION
+  udf_bitmask_lowest_7() AS (0x7F);
+  --
+CREATE TEMP FUNCTION
+  udf_shift_one_day(x INT64) AS (IFNULL((x << 1) & udf_bitmask_lowest_28(),
+      0));
+  --
+CREATE TEMP FUNCTION
+  udf_combine_days(prev INT64,
+    curr INT64) AS (udf_shift_one_day(prev) + IFNULL(curr,
+      0));
+  --
+WITH
+  _current AS (
+  SELECT
+    -- In this raw table, we capture the history of activity over the past
+    -- 28 days for each usage criterion as a single 64-bit integer. The
+    -- rightmost bit represents whether the user was active in the current day.
+    CAST(TRUE AS INT64) AS days_seen_bits,
+    -- We only trust profile_date if it is within one week of the ping submission,
+    -- so we ignore any value more than seven days old.
+    IFNULL(1 << DATE_DIFF(submission_date, profile_date, DAY), 0) & udf_bitmask_lowest_7() AS days_created_profile_bits,
```

Comment:
The `IFNULL` here is redundant with the `COALESCE` below
```suggestion
    1 << DATE_DIFF(submission_date, profile_date, DAY) & udf_bitmask_lowest_7() AS days_created_profile_bits,
```

Reply:
My thought here was to consolidate logic such that all the `days_*_bits` columns defined for `_current` are non-nullable.

Then we don't need the trailing 0 in the COALESCE, because we're guaranteed that one of the two values will be non-null. If this seems less clear to you, though, I'll change to match your suggestions.

Reply:
your plan wfm